src/API/Google/Settings.php

```diff
@@ -155,12 +202,33 @@ protected function get_times(): array {
 	 *
 	 * @return array
 	 */
-	protected function get_rates(): array {
+	protected function get_shipping_rates_from_database(): array {
 		$rate_query = $this->container->get( RateQuery::class );
 
 		return $rate_query->get_results();
 	}
 
+	/**
+	 * Get shipping rate data from WooCommerce shipping settings.
+	 *
+	 * @return array
+	 */
+	protected function get_shipping_rates_from_woocommerce(): array {
+		/** @var TargetAudience $target_audience */
+		$target_audience  = $this->container->get( TargetAudience::class );
+		$target_countries = $target_audience->get_target_countries();
+		/** @var ShippingZone $shipping_zone */
+		$shipping_zone = $this->container->get( ShippingZone::class );
+
+		$rates = [];
+		foreach ( $target_countries as $country ) {
+			$country_rates = $shipping_zone->get_shipping_rates_for_country( $country );
+			$rates         = array_merge( $rates, $country_rates );
+		}
+
+		return $rates;
```

Comment:
❓ 

When I try to sync the shipping rates for Austria I get the following error:

```
{
    "code": "gla_setting_sync_error",
    "message": "{\n  \"error\": {\n    \"code\": 400,\n    \"message\": \"[services[0].deliveryTime] Missing required service delivery time\",\n    \"errors\": [\n      {\n        \"message\": \"[services[0].deliveryTime] Missing required service delivery time\",\n        \"domain\": \"global\",\n        \"reason\": \"required\"\n      }\n    ]\n  }\n}\n",
    "data": {
        "status": 500,
        "message": "",
        "data": {
            "error": {
                "code": 400,
                "message": "[services[0].deliveryTime] Missing required service delivery time",
                "errors": [
                    {
                        "message": "[services[0].deliveryTime] Missing required service delivery time",
                        "domain": "global",
                        "reason": "required"
                    }
                ]
            }
        }
    }
}
```
I think Google requires deliveryTime for some countries?

Reply:
deliveryTime is required but that should already be set if you go through the onboarding wizard. Can you edit your free-listing settings or go through onboarding once again and see if the issue will be resolved? It should be saved under the `gla_merchant_center` option in the `wp_options` table with the key `shipping_time`.